### PR TITLE
ACI-CNI 5.2.3.2 upgrade 

### PR DESCRIPTION
**Pull request by siva-muni**
- Add ACI template
- Add ACI CNI template and images.

### CODE REVIEW

File: go.mod
Comment (snasovich) [inline go.mod]: Is this a temporary change until RKE changes in https://github.com/rancher/rke/pull/2963/files are merged?

File: go.mod
Comment (snasovich) [inline go.mod]: Is this a temporary change until RKE changes in https://github.com/rancher/rke/pull/2963/files are merged?

File: rke/k8s_rke_system_images.go
Comment (snasovich) [inline rke/k8s_rke_system_images.go]: Updating existing already released versions is not supported. A new version (like `v1.20.15-rancher1-5`) needs to be introduced with bumped ACI versions.

File: rke/k8s_rke_system_images.go
Comment (snasovich) [inline rke/k8s_rke_system_images.go]: What is the reason for Nginx controller version downgrade?

File: rke/k8s_rke_system_images.go
Comment (snasovich) [inline rke/k8s_rke_system_images.go]: Since `v1.21.13-rancher1-1` based on k8s 1.21.13 is already released, we would generally not introduce `v1.21.12-rancher1-2` based on not the latest patch. Is there a reason this updated ACI can't be based on 1.21.13?

In fact, for the upcoming release we're introducing k8s June patch versions - https://github.com/rancher/kontainer-driver-metadata/pull/921. So realistically we would want to introduce updated ACI images in the versions to be added per that PR: `1.23.8-rancher1-1`, `1.22.11-rancher1-1`, `1.21.14-rancher1-1`. The best option would be to wait for the linked PR to be merged and then rebase your PR and make updates to `1.23.8-rancher1-1`, `1.22.11-rancher1-1`, `1.21.14-rancher1-1`. In this case making updates to already existing versions is an acceptable process as these versions were not released yet (not merged to `release-v2.6` branch).

File: rke/k8s_rke_system_images.go
Comment (snasovich) [inline rke/k8s_rke_system_images.go]: Same as one of the previous comments - looks like old version of Ingress Controller is referenced. I did not check other images yet, please make sure you're not inadvertently downgrading them.

File: rke/k8s_rke_system_images.go
Comment (snasovich) [inline rke/k8s_rke_system_images.go]: New 1.22 version needs to be defined further down after the latest already defined 1.22 - we want them to be sorted.

File: rke/templates/aci.go
Comment (snasovich) [inline rke/templates/aci.go]: Updating the existing template is not supported as will it be used by older versions of ACI that do not support it. You can see an example of adding new template (for Ingress NGINX controller) in this PR https://github.com/rancher/kontainer-driver-metadata/pull/889 and follow the same approach.

File: rke/k8s_rke_system_images.go
Comment (snasovich) [inline rke/k8s_rke_system_images.go]: Is there a plan to update ACI version for 1.23 version too? Or it's not supported yet?

File: go.mod
Comment (snasovich) [inline go.mod]: Is there a specific need to bump `logrus` here?

**Review by snasovich — CHANGES_REQUESTED** [review]
@siva-muni, thank you for submitting this PR.

I've left a few specific comments to address.
Additionally, here are couple general comments:
* The build is failing, please look into why and address: https://github.com/rancher/kontainer-driver-metadata/pull/889/commits
* To merge this PR, we want it to have 2 commits - actual changes and `go generate` (so updates to `data.json`) separately.

**Comment by siva-muni** [comment]
[snasovich](https://github.com/snasovich), Thanks for the review, we will address your comments.

**Comment by siva-muni** [comment]
We addressed the comments in [91eb1fe](https://github.com/rancher/kontainer-driver-metadata/pull/922/commits/91eb1fef675132f4c1205da6c58b063f44e9759a), [07b240a](https://github.com/rancher/kontainer-driver-metadata/pull/922/commits/07b240a1fcf1f583c2baa6598e05d080a2c779bf), but still we need to test this code once it works fine, We will update, we are working on this.

**Comment by siva-muni** [comment]
With these changes, we are seeing the issues, We will fix and update

**Comment by siva-muni** [comment]
Made few more changes, testing in progress.

File: rke/templates/aci.go
Comment (snasovich) [inline rke/templates/aci.go]: Please put this new template in a separate file to follow the same pattern as used by other templates that have multiple versions (e.g. nginx, see files in https://github.com/rancher/kontainer-driver-metadata/tree/dev-v2.6/rke/templates).

File: rke/templates/templates.go
Comment (snasovich) [inline rke/templates/templates.go]: A lot more conditions need to be introduced here to ensure that updated `aciv523` template is used for newly introduced versions only. With the current condition (`>=1.20.0-rancher0 <1.23.0-rancher1-1`) it will be used for any k8s version starting with 1.20, but previously existing versions (e.g. `v1.20.15-rancher1-4`) reference 5.1.1.0 images and thus cannot use the new template that is built for 5.2.3.2.
Please see below an example for `kdm.NginxIngress` that demonstrates more granular conditions to ensure only newly added versions use new template.

File: rke/k8s_rke_system_images.go
Comment (snasovich) [inline rke/k8s_rke_system_images.go]: This is out of order with 1.21 version following 1.22. This will be a moot point once we rebase (see overall review comment).

**Review by snasovich — CHANGES_REQUESTED** [review]
I see most of my previous comments are addressed, thank you.

Added some more comments to address.

Also, note that per https://github.com/rancher/kontainer-driver-metadata/pull/922#discussion_r906095796 the best option would be to rebase after PR #921 is merged (should be very soon) to update versions it added with new ACI images /templates instead of introducing new ones. It should be merged very soon, likely today.

**Comment by snasovich** [comment]
@siva-muni , #921 is now merged so please go ahead with rebasing this PR and updating versions added by #921 instead of introducing new ones.

**Comment by siva-muni** [comment]
[snasovich](https://github.com/snasovich), thanks for the review, I will address the comments

File: rke/templates/aci.go
Comment (siva-muni) [reply rke/templates/aci.go]: Added separate file for new template in [7e7eabb](https://github.com/rancher/kontainer-driver-metadata/pull/922/commits/7e7eabbba1b4688c0359d90daf1072e09a1fe64f)

File: rke/templates/templates.go
Comment (siva-muni) [reply rke/templates/templates.go]: Updated required conditions in [7e7eabb](https://github.com/rancher/kontainer-driver-metadata/pull/922/commits/7e7eabbba1b4688c0359d90daf1072e09a1fe64f)

**Comment by siva-muni** [comment]
Addressed the comments, We have updated the images in the  versions added by https://github.com/rancher/kontainer-driver-metadata/pull/921 
 - 1.22.11-rancher1-1	
- 1.21.14-rancher1-1	
- 1.20.15-rancher2-1

**Comment by siva-muni** [comment]
With these changes, testing is in-progress

File: rke/k8s_rke_system_images.go
Comment (siva-muni) [reply rke/k8s_rke_system_images.go]: removed this change

**Review by snasovich — DISMISSED** [review]
@siva-muni , changes LGTM, thank you for addressing my comments.

Please report the results of your testing and once you confirm it's successful and we have one more approval, this can be merged.

cc: @kinarashah

**Comment by siva-muni** [comment]
[snasovich](https://github.com/snasovich), We are finished our testing, it was successful.  We are not seeing any issues.

**Comment by kinarashah** [comment]
@siva-muni Is there any reason we don't want the images to be updated for 1.23.x and 1.24.x? If users upgrade from 1.22.x to 1.23.x aci version would downgrade, is that expected? Apologies for missing necessary context if it's obvious.

**Comment by siva-muni** [comment]
[kinarashah](https://github.com/kinarashah),  we added the new images now for 1.23 and 1.24 versions as well. This is an oversight from our side as we are doing this for the first time.

**Comment by snasovich** [comment]
@siva-muni , thank you for following up on this. Are we waiting for additional testing or good to merge once @kinarashah approves?

**Comment by siva-muni** [comment]
[snasovich](https://github.com/snasovich), We are done with testing. We are good to merge.

**Review by snasovich — APPROVED** [review]
@siva-muni , thank you for updating the PR and reporting that the testing is successful. LGTM.
@kinarashah , could you please take a peek at it as well and if it looks good to you, please merge it.

**Review by kinarashah — APPROVED** [review]

**Pull request merged**
